### PR TITLE
added initial port crd page

### PR DESCRIPTION
**Pull request by MatanHeledPort**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

## Added docs pages

Please also include the path for the added docs

- Quickstart (`/`)
- Blueprint (`/platform-overview/port-components/blueprint`)
- ...

## Updated docs pages

Please also include the path for the updated docs

- Quickstart (`/`)
- Blueprint (`/platform-overview/port-components/blueprint`)
- ...

### CODE REVIEW

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
+---
+sidebar_position: 7
+---
+
+# Entity CRD
+
+Port's K8s exporter allows exporting data from any resource in your Kubernetes clusters, including [CustomResourceDefinitions](port-crd.md)(CRDs).
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:7]:
```suggestion
Port's K8s exporter allows exporting data from any resource in your Kubernetes clusters, including  [CustomResourceDefinitions](port-crd.md)(CRDs).
```

Comment (MPTG94) [reply docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:7]: Link is wrong? I think it should either take to the page in our docs about ingesting custom CRDs, or to the K8s docs page about CRDs

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+To take advantage of the flexibility of Port's K8s exporter, Port provides additional CRDs which allow to use K8s resource definitions as a source of [entities](../../sync-data-to-catalog/sync-data-to-catalog.md#creating-entities) in your software catalog.
+
+:::tip
+All CRDs provided by Port can be found [here.](https://github.com/port-labs/port-crds)
+:::
+
+# Port CRDs
+
+A Port entity can represent any kind of data in your infrastructure, from nodes to pods, and even external data such as repositories. This means Port entities can reference both cluster-scoped Kubernetes resources, and namespace-scoped Kubernetes resources. To achieve this, 2 CRDs are provided:
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:16]: The cluster-scoped and namespace-scoped terminology makes it seem like this is meant to be used only for K8s related objects, but our goal here is to explain that it is possible to ingest any sort of information using GitOps, it is only a matter of choosing the correct CRD scope for it.

So the way the sentence is written should be altered slightly

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+              properties:
+                type: object
+                x-kubernetes-preserve-unknown-fields: true
+              relations:
+                type: object
+                x-kubernetes-preserve-unknown-fields: true
+            required:
+              - blueprint
+              - identifier
+  scope: Namespaced
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:67]: The d is intended?

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+  identifier: "<Entity ID>"
+  properties:
+    myStringProp: string
+    myArrayProp:
+      - string_1
+      - string_2
+      - string_3
+    myUrlProp: https://test-url.com
+  relations:
+    singleRelation: relation-target-id
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:168]:
```suggestion
    mySingleRelation: relation-target-id
```

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    multipleRelations:
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:169]:
```suggestion
    myManyRelation:
```

**Review by MPTG94 — COMMENTED** [review]
Added some initial comments

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md (hunk 3 above)
Comment (MatanHeledPort) [reply docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:67]: yes, this is the format

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md (hunk 2 above)
Comment (MatanHeledPort) [reply docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md]: Changed this, LMK what you think

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md (hunk 1 above)
Comment (MatanHeledPort) [reply docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md]: gonna fix

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+# Run this to install Port's namespace-scoped CRD
+kubectl apply -f https://raw.githubusercontent.com/port-labs/port-crds/main/port-entity-crd-namespace.yaml
+
+# Run this to install Port's cluster-scoped CRD
+kubectl apply -f https://raw.githubusercontent.com/port-labs/port-crds/main/port-entity-crd-cluster.yaml
+```
+
+## Exporting Port's custom resources
+
+To export the Port entity CRDs using Port's K8s exporter, you will need to create custom configuration for your [config.yaml](./kubernetes.md#exporter-configyml-file). This mapping configuration will match the blueprint data model you defined in your software catalog.
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:214]:
```suggestion
To export the Port entity CRDs using Port's K8s exporter, you will need to create a custom configuration for your [config.yaml](./kubernetes.md#exporter-configyml-file). This mapping configuration will match the blueprint data model you defined in your software catalog.
```

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+
+:::tip
+All CRDs provided by Port can be found [here.](https://github.com/port-labs/port-crds)
+:::
+
+# Port CRDs
+
+A Port entity can represent any kind of data in your infrastructure, from nodes to pods to, to non-Kuberenetes related entities such as repositories or microservices. To achieve this level of abstraction, 2 CRDs are provided:
+
+**Namespace scoped entity CRD** - `getport.io/v1/Entity`
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:18]: Consider changing these bolded sentences to headings (use pounds - `#`)

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+To learn how to use Port CRDs to fit your needs, you will follow an example. It will give you a general understanding of how to map any data you would like.
+
+### Example - Mapping a microservice using Port CRDs
+
+The goal for this example is to map a microservice using Port's CRD and Port's K8s exporter. For this example, you will map [Port's docs microservice](https://github.com/port-labs/port-docs) as a Port entity.
+
+:::note Prerequisites
+Before getting started:
+
+- Prepare your [Port credentials](../../../build-your-software-catalog/sync-data-to-catalog/api/api.md#find-your-port-credentials)
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:225]:
```suggestion
- Prepare your [Port credentials](../../../build-your-software-catalog/sync-data-to-catalog/api/api.md#find-your-port-credentials);
```

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md [hunk 9]
```diff
@@ -0,0 +1,340 @@
...
+- Be familiar with [Port's K8s exporter](./kubernetes.md) and configuration
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:226]:
```suggestion
- Be familiar with [Port's K8s exporter](./kubernetes.md) and configuration;
```

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md [hunk 10]
```diff
@@ -0,0 +1,340 @@
...
+- Make sure you are connected to a K8s cluster using `kubectl`
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:227]:
```suggestion
- Make sure you are connected to a K8s cluster using `kubectl`.
```

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md [hunk 11]
```diff
@@ -0,0 +1,340 @@
...
+
+:::
+
+1. **Deploy the Port CRD** - Follow the [deployment step](./port-crd.md#deploying-ports-crds) to deploy the Port CRD. You will only need the cluster-scoped entity CRD.
+
+2. **Creating the blueprint** - You will begin by defining the blueprint which will represent a microservice in your software catalog.
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:233]:
```suggestion
2. **Create the blueprint** - You will begin by defining the blueprint which will represent a microservice in your software catalog.
```

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md [hunk 12]
```diff
@@ -0,0 +1,340 @@
...
+:::note Prerequisites
+Before getting started:
+
+- Prepare your [Port credentials](../../../build-your-software-catalog/sync-data-to-catalog/api/api.md#find-your-port-credentials)
+- Be familiar with [Port's K8s exporter](./kubernetes.md) and configuration
+- Make sure you are connected to a K8s cluster using `kubectl`
+
+:::
+
+1. **Deploy the Port CRD** - Follow the [deployment step](./port-crd.md#deploying-ports-crds) to deploy the Port CRD. You will only need the cluster-scoped entity CRD.
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:231]:
```suggestion
1. **Deploy the Port CRD** - follow the [deployment step](./port-crd.md#deploying-ports-crds) to deploy the Port CRD. You will only need the cluster-scoped entity CRD.
```

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md [hunk 13]
```diff
@@ -0,0 +1,340 @@
...
+    },
+    "required": []
+  },
+  "mirrorProperties": {},
+  "calculationProperties": {},
+  "relations": {}
+}
+```
+
+3. **Create a Port Entity custom resource in your cluster** - Create an Entity CR which will represent a microservice, using the scheme defined in your blueprint:
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:280]:
```suggestion
3. **Create a Port entity custom resource in your cluster** - create an Entity CR which will represent a microservice, using the scheme defined in your blueprint:
```

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md [hunk 14]
```diff
@@ -0,0 +1,340 @@
...
+  identifier: my-awesome-microservice
+  properties:
+    description: This entity represents my awesome microservice
+    onCall: some@email.com
+    slackChannel: https://domain.slack.com/archives/12345678
+    datadogUrl: https://www.datadoghq.com/
+    language: typescript
+```
+
+2.  Apply CRD manifest to your cluster:
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:301]:
```suggestion
2.  Apply the CRD manifest to your cluster:
```

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md [hunk 15]
```diff
@@ -0,0 +1,340 @@
...
+
+```bash showLineNumbers
+kubectl apply -f port-entity.yaml
+```
+
+4. **Create mapping configuration for the K8s exporter** - Create (or add to existing) the following `config.yaml` configuration for mapping this CRD using Port's k8s exporter:
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:307]:
```suggestion
4. **Create a mapping configuration for the K8s exporter** - create (or add to an existing) the following `config.yaml` configuration to map this CRD using Port's k8s exporter:
```

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md (hunk 14 above)
Comment (MPTG94) [reply docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:301]: The numbering here breaks, probably due to some indentation issues

File: docs/build-your-software-catalog/sync-data-to-catalog/git/gitops-using-port-crd.md [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+# GitOps using Port CRDs
+
+You can use GitOps, [Port's Entity CRDs](../kubernetes/port-crd.md) and [Port's K8s exporter](../kubernetes/kubernetes.md) to export custom entities in to Port.
+
+:::note
+For fully understanding how to use GitOps with Port's CRDs to map entities, make sure you are familiar with:
+
+- [Ports Kubernetes exporter](../kubernetes/kubernetes.md)
+- [Ports Entity CRDs](../kubernetes/port-crd.md)
+  :::
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/git/gitops-using-port-crd.md:14]: Formatting issue, the entire page is part of the admonition block

File: docs/build-your-software-catalog/sync-data-to-catalog/git/gitops-using-port-crd.md [hunk 17]
```diff
@@ -0,0 +1,35 @@
+---
+sidebar_position: 4
+---
+
+# GitOps using Port CRDs
+
+You can use GitOps, [Port's Entity CRDs](../kubernetes/port-crd.md) and [Port's K8s exporter](../kubernetes/kubernetes.md) to export custom entities in to Port.
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/git/gitops-using-port-crd.md:7]: flip the order between port CRD and K8s exporter

File: docs/build-your-software-catalog/sync-data-to-catalog/git/gitops-using-port-crd.md [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+For fully understanding how to use GitOps with Port's CRDs to map entities, make sure you are familiar with:
+
+- [Ports Kubernetes exporter](../kubernetes/kubernetes.md)
+- [Ports Entity CRDs](../kubernetes/port-crd.md)
+  :::
+
+## Export entities with GitOps and Kubernetes use cases
+
+- Use your K8s clusters as the source-of-truth for your **microservices**, **packages**, **libraries** and other software catalog assets;
+- Update Port in a "Push Only" method, where no elevated permissions are required for Port to interact with your infrastructure.
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/git/gitops-using-port-crd.md:19]:
```suggestion
- Update Port in a "Push Only" method, where no elevated permissions are required for Port to interact with your infrastructure;
```

File: docs/build-your-software-catalog/sync-data-to-catalog/git/gitops-using-port-crd.md [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+- Allow developers to keep the catalog up-to-date, by making updates to Kubernetes manifest files in their Git repositories;
+- Create a standardized way to document software catalog assets in your organization;
+- etc.
+
+## Managing entities defined using CRDs and GitOps
+
+Port's CRDs allow defining and mapping any kind of entity using Kubernetes. You can find an example for doing this in the [Port CRDs - mapping a microservice example](../kubernetes/port-crd.md#example---mapping-a-microservice-using-port-crds).
+Mapping entities can be done using any Continuous Deployment(CD) solution, for example ArgoCD or FluxCD, by deploying the Port custom resources using the CD solution, and exporting them to Port
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/git/gitops-using-port-crd.md:27]:
```suggestion
Mapping entities can be done using any Continuous Deployment (CD) solution, for example ArgoCD or FluxCD, by deploying the Port custom resources using the CD solution, and mapping their definition to Port using Port's K8s exporter.
```

Comment (MPTG94) [reply docs/build-your-software-catalog/sync-data-to-catalog/git/gitops-using-port-crd.md:27]: Add spacing between the 2 sentences

File: docs/build-your-software-catalog/sync-data-to-catalog/git/gitops-using-port-crd.md [hunk 20]
```diff
@@ -0,0 +1,35 @@
...
+
+To achieve this, follow these general steps for any CD solution:
+
+1. Navigate to [Port's CRDs document](../kubernetes/port-crd.md#deploying-ports-crds) page to learn how to deploy the CRDs. You can deploy then as part of your CD pipeline by placing the CRD manifests in one of your CD source directory/applications.
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/git/gitops-using-port-crd.md:31]:
```suggestion
1. Navigate to [Port's CRDs document](../kubernetes/port-crd.md#deploying-ports-crds) page to learn how to deploy the CRDs. You can deploy them as part of your CD pipeline by placing the CRD manifests in one of your CD source directory/applications;
```

File: docs/build-your-software-catalog/sync-data-to-catalog/git/gitops-using-port-crd.md [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+2. Define a Kubernetes Port entity manifest using Port's CRD, which contains the data model and data you wish to map to Port, and deploy it to your kubernetes cluster using your CD solution.
+3. Update Port's K8s exporter `config.yaml` to map the Port CRD you just created. Deploy the new `config.yaml`.
+
+The entities should appear in your Port environment.
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/git/gitops-using-port-crd.md:35]:
```suggestion
The entities defined using Port's CRD will appear in your Port environment.
```

File: docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md [hunk 22]
```diff
@@ -0,0 +1,340 @@
...
+
+   ```bash showLineNumbers
+   export CLUSTER_NAME="<YOUR_CLUSTER_NAME>" # Defaults to 'my-cluster'
+   export CONFIG_YAML_URL="<PATH_TO_CONFIG_YAML>/config.yaml"
+   export PORT_CLIENT_ID="<PORT_CLIENT_ID>"
+   export PORT_CLIENT_SECRET="<PORT_CLIENT_SECRET>"
+   curl -s https://raw.githubusercontent.com/port-labs/template-assets/main/kubernetes/install.sh | bash
+   ```
+
+You should now be able to see the newly exported entity in your Port environment.
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/kubernetes/port-crd.md:340]:
```suggestion
You will now be able to see the newly exported entity in your Port environment.
```

**Review by MPTG94 — APPROVED** [review]
Fix comments and merge

**Pull request merged**